cmd/kubelet/app/server.go

```diff
@@ -1225,6 +1225,7 @@ func setContentTypeForClient(cfg *restclient.Config, contentType string) {
 //
 // Eventually, #2 will be replaced with instances of #3
 func RunKubelet(ctx context.Context, kubeServer *options.KubeletServer, kubeDeps *kubelet.Dependencies) error {
+	logger := klog.FromContext(ctx)
```

Comment:
I'm not sure passing logger instead of `context` is a good idea for this module overall, not just for this function. If we do that we'll have to use `context.Background()` down the pass as this module is responsible for running the rest of `Kubelet`. Please, consider passing `context` instead of `logger`.

Reply:
Yes, it's simpler, but it's not what we do in this set of PRs. My overall idea was to pass upper level context to the components, so that they can use it either now or later. Passing logger is ok for functions where the context is not used and is not **going** to be used.